Grbl_Esp32/src/GCode.cpp

```diff
@@ -197,113 +193,165 @@ uint8_t gc_execute_line(char* line, uint8_t client) {
             case 'G':
                 // Determine 'G' command and its modal group
                 switch (int_value) {
-                    case 10:
-                    case 28:
-                    case 30:
+                    // Modal Group G0 - non-modal actions
+                    case 10: gc_block.non_modal_command = NonModal::SetCoordinateData; goto check_mantissa;
+                    case 28: gc_block.non_modal_command = mantissa ? NonModal::SetHome0 : NonModal::GoHome0; goto check_mantissa;
+                    case 30: gc_block.non_modal_command = mantissa ? NonModal::SetHome1 : NonModal::GoHome1; goto check_mantissa;
                     case 92:
+                        gc_block.non_modal_command = mantissa ? NonModal::ResetCoordinateOffset : NonModal::SetCoordinateOffset;
+                    check_mantissa:
                         // Check for G10/28/30/92 being called with G0/1/2/3/38 on same block.
                         // * G43.1 is also an axis command but is not explicitly defined this way.
-                        if (mantissa == 0) {  // Ignore G28.1, G30.1, and G92.1
-                            if (axis_command) {
-                                FAIL(STATUS_GCODE_AXIS_COMMAND_CONFLICT);  // [Axis word/command conflict]
-                            }
-                            axis_command = AXIS_COMMAND_NON_MODAL;
+                        switch (mantissa) {
+                            case 0:  // Ignore G28.1, G30.1, and G92.1
+                                if (axis_command != AxisCommand::None) {
+                                    FAIL(STATUS_GCODE_AXIS_COMMAND_CONFLICT);  // [Axis word/command conflict]
+                                }
+                                axis_command = AxisCommand::NonModal;
+                                break;
+                            case 10:
```

Comment:
The '10' and 'default' cases are not in the original code for int_value == 10. The '0' case is.

Reply:
I restructured that code to handle G10 according to the original logic, then added a variety of test cases in tests/parser.nc to look at various combinations.  It passes the tests, in that the new code generates identical responses to the old code.